src/main/java/com/github/quiram/buildhotspots/visualisation/BuildConfiguration.java

```diff
@@ -63,34 +117,51 @@ private double scalePercentage(byte p_percentage) {
     private double m_origLayoutY = -1;
 
     public int getDepth() {
-        return m_relatedDependencies.stream().map(d -> {
+        return m_Dependencies.stream().map(d -> {
             BuildConfiguration origin = d.getOrigin();
             int depth = 0;
-            if (origin != this) {
-                depth = 1 + origin.getDepth();
-            }
+            depth = 1 + origin.getDepth();
```

Comment:
I think this check is needed to avoid infinite loops because both ends of the dependency (origin and target) hold a reference to the dependency. That means that the origin can get a dependency, find the origin (which is itself) and recursively call for getDepth().